core-common/src/main/java/eu/arrowhead/common/processor/NetworkAddressPreProcessor.java

```diff
@@ -0,0 +1,168 @@
+/********************************************************************************
+ * Copyright (c) 2021 AITIA
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   AITIA - implementation
+ *   Arrowhead Consortia - conceptualization
+ ********************************************************************************/
+
+package eu.arrowhead.common.processor;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.stereotype.Component;
+
+import eu.arrowhead.common.Utilities;
+
+@Component
+public class NetworkAddressPreProcessor {
+	
+	private final Logger logger = LogManager.getLogger(NetworkAddressPreProcessor.class);
+	
+	//=================================================================================================
+	// methods
+
+	//-------------------------------------------------------------------------------------------------
+	public String normalize(final String address) {
+		logger.debug("normalize started...");
+		
+		if (Utilities.isEmpty(address)) {
+			return "";
+		}
+		final String candidate = address.trim();
+		
+		// Simple string
+		if (!candidate.contains(".") && !candidate.contains(":")) {
+			return candidate;
+		
+		// Possible IPv4 or domain name
+		} else if(candidate.contains(".") && !candidate.contains(":")) {
+			return candidate;
+		
+		// Possible IPv6
+		} else if (!candidate.contains(".") && candidate.contains(":")) {
+			return normalizeIPv6(candidate);
+			
+		// Possible IPv6-IPv4 hybrid
+		} else {
+			return normalizeIPv6IPv4Hybrid(candidate);
+		}
+	}
+	
+	//=================================================================================================
+	// assistant methods
+
+	//-------------------------------------------------------------------------------------------------
+	private String normalizeIPv6(final String candidate) {
+		logger.debug("normalizeIPv6 started...");
+		final List<String> groups = new ArrayList<>(8);
+		
+		final String[] split = candidate.split(":");
+		for (final String part : split) {
+			String group = part;
+			if (group.length() < 4) {
+				for (int i = 0; i < (4 - part.length()); i++) {
+					group = "0" + group;
+				}
+			} 
+			groups.add(group);
+		}
+		
+		if (candidate.startsWith("::") && !candidate.endsWith("::")) {
```

Comment:
minor issue: maybe double colon should be a constants too

Reply:
I know, that IPv6 protocol allows only one double colon, but I think we are able to interpret more occurance till the address is not starts and ends with at the same time. So if an address is coming from a mgmt endpoint and the operator use such incorrect, but widespread abbreviation, than we still can handle it.

abbreviation |  
-- | --




abbreviation |  
-- | --




abbreviation |  
-- | --